File: sdk-android-library/src/main/java/com/xyfindables/sdk/actionHelper/XYGPSInterval.java
package com.xyfindables.sdk.actionHelper;

import android.bluetooth.BluetoothGatt;
import android.bluetooth.BluetoothGattCharacteristic;
import android.util.Log;

import com.xyfindables.core.XYBase;
import com.xyfindables.sdk.XYDevice;
import com.xyfindables.sdk.action.XYDeviceActionGetGPSInterval;
import com.xyfindables.sdk.action.XYDeviceActionSetGPSInterval;

/**
 * Created by alex.mcelroy on 9/29/2017.
 */

public class XYGPSInterval extends XYActionHelper {

    private static final String TAG = XYGPSInterval.class.getSimpleName();

    public interface Callback extends XYActionHelper.Callback {
        void started (boolean success, byte[] value);
    }

    public XYGPSInterval(XYDevice device, final Callback callback) {
        action = new XYDeviceActionGetGPSInterval(device) {
            @Override
            public boolean statusChanged(int status, BluetoothGatt gatt, BluetoothGattCharacteristic characteristic, boolean success) {
                Log.v(TAG, "statusChanged:" + status + ":" + success);
                boolean result = super.statusChanged(status, gatt, characteristic, success);
                switch (status) {
                    case STATUS_CHARACTERISTIC_READ:
                        callback.started(success, value);
                        break;
                    case STATUS_COMPLETED:
                        callback.completed(success);
                        break;
                }
                return result;
            }
        };
    }

    public XYGPSInterval(XYDevice device, byte[] value, final Callback callback) {
        action = new XYDeviceActionSetGPSInterval(device, value) {
            @Override
            public boolean statusChanged(int status, BluetoothGatt gatt, BluetoothGattCharacteristic characteristic, boolean success) {
                Log.v(TAG, "statusChanged:" + status + ":" + success);
                boolean result = super.statusChanged(status, gatt, characteristic, success);
                switch (status) {
                    case STATUS_COMPLETED:
                        callback.completed(success);
                        break;
                }
                return result;
            }
        };
    }
}

File: sdk-android-library/src/main/java/com/xyfindables/sdk/actionHelper/XYGPSInterval.kt
package com.xyfindables.sdk.actionHelper

import android.bluetooth.BluetoothGatt
import android.bluetooth.BluetoothGattCharacteristic
import com.xyfindables.sdk.XYDevice
import com.xyfindables.sdk.action.XYDeviceAction
import com.xyfindables.sdk.action.XYDeviceActionGetGPSInterval
import com.xyfindables.sdk.action.XYDeviceActionSetGPSInterval

/**
 * Created by alex.mcelroy on 9/29/2017.
 */

class XYGPSInterval : XYActionHelper {

    interface Callback : XYActionHelper.Callback {
        fun started(success: Boolean, value: ByteArray)
    }

    constructor(device: XYDevice, callback: Callback) {
        action = object : XYDeviceActionGetGPSInterval(device) {
            override fun statusChanged(status: Int, gatt: BluetoothGatt?, characteristic: BluetoothGattCharacteristic?, success: Boolean): Boolean {
                logExtreme(TAG, "statusChanged:$status:$success")
                val result = super.statusChanged(status, gatt, characteristic, success)
                when (status) {
                    XYDeviceAction.STATUS_CHARACTERISTIC_READ -> callback.started(success, value!!)
                    XYDeviceAction.STATUS_COMPLETED -> callback.completed(success)
                }
                return result
            }
        }
    }

    constructor(device: XYDevice, value: ByteArray, callback: Callback) {
        action = object : XYDeviceActionSetGPSInterval(device, value) {
            override fun statusChanged(status: Int, gatt: BluetoothGatt?, characteristic: BluetoothGattCharacteristic?, success: Boolean): Boolean {
                logExtreme(TAG, "statusChanged:$status:$success")
                val result = super.statusChanged(status, gatt, characteristic, success)
                when (status) {
                    XYDeviceAction.STATUS_COMPLETED -> callback.completed(success)
                }
                return result
            }
        }
    }

    companion object {

        private val TAG = XYGPSInterval::class.java.simpleName
    }
}
